### ISSUE TITLE
isatab.validate2 - Some exceptions that occurs when validating are hidden

### ISSUE THREAD

**Issue by yjarosz** [ISA-tools/isa-api#171]
Hello,
when trying to validate isatab files I had some troubles. Basically I tried with a simple investigation file that I was sure it must have failed, but not errors was show on the logs:

``` 
➜  datamanager python tests/test_isa.py                                                                                                                                                08:24:07 (master) ✗
2016-11-30 20:26:11,317 INFO: ISA tab Validator from ISA tools API v0.3
2016-11-30 20:26:11,317 INFO: Loading... /var/folders/gn/p47pzlx16v7bnwnwv4b10rv9skqz7l/T/tmpc8drdmmd
2016-11-30 20:26:11,317 INFO: Loading ONTOLOGY SOURCE REFERENCE section
{'validation_finished': True, 'errors': [], 'warnings': []}
```

I made a small snippet that can reproduce the error:
```
def test_validate_isatab():
    import tempfile
    import os
    with tempfile.NamedTemporaryFile(delete=False, mode='w') as fp:
        fp.write(u"""This    one    should fail
        And give errors
        """)
        name = fp.name
    with open(name, 'r') as fp:
        print(isatab.validate2(fp))
    os.remove(name)
```

Basically from [this block](https://github.com/ISA-tools/isa-api/blob/master/isatools/isatab.py#L2615) to [this one](https://github.com/ISA-tools/isa-api/blob/master/isatools/isatab.py#L2639) only `CParserError , ValueError or SystemError` are got and propagated to the logger. Every other ones get hidden. By adding 

```
    except Exception as se:
        errors.append({
            "message": "Unknown/System Error",
            "supplemental": "The validator could not identify what the error is: {}".format(str(se)),
            "code": 0
        })
        logger.fatal("(F) Something went very very wrong! :(")
        logger.fatal(se)
``` 

just before the finally block, it can bubble up the error to the log:

```
➜  datamanager python tests/test_isa.py                                                                                                                                                08:26:11 (master) ✗
2016-11-30 20:32:51,668 INFO: ISA tab Validator from ISA tools API v0.3
2016-11-30 20:32:51,668 INFO: Loading... /var/folders/gn/p47pzlx16v7bnwnwv4b10rv9skqz7l/T/tmp9_lq07cr
2016-11-30 20:32:51,668 INFO: Loading ONTOLOGY SOURCE REFERENCE section
2016-11-30 20:32:51,668 CRITICAL: (F) Something went very very wrong! :(
2016-11-30 20:32:51,668 CRITICAL: Expected: ONTOLOGY SOURCE REFERENCE section, but got: This    one    should fail
{'errors': [{'supplemental': 'The validator could not identify what the error is: Expected: ONTOLOGY SOURCE REFERENCE section, but got: This    one    should fail', 'message': 'Unknown/System Error', 'code': 0}], 'validation_finished': True, 'warnings': []}
```

To me, I get a more meaningful message leading me to the error :)
Considering the fact that I'm new to ISA-Tab, I assume that I will make a lot of mistakes
and it would be more helpful to get the error message.

Not sure why those Exceptions get completely silenced though.

**Comment by yjarosz** [comment]
BTW I can make a pull request if needed and/or include this snippet maybe in [test_validate_sample_data.py](https://github.com/ISA-tools/isa-api/blob/master/tests/test_validate_sample_data.py)?

**Comment by djcomlab** [comment]
Hi @yjarosz thanks for this, please feel free to make pull requests as we're happy for contributions of fixes!

**Comment by djcomlab** [comment]
Fixed by PR #173, closing
